lib/Marlin/Marlin/src/feature/prusa/MMU2/mmu2.cpp

```diff
@@ -346,14 +346,13 @@ bool MMU2::rx_str_P(const char *str) {
     uint8_t i = strlen(rx_buffer);
 
     while (mmuSerial.available()) {
-        rx_buffer[i++] = mmuSerial.read();
-        rx_buffer[i] = '\0';
-
-        if (i == sizeof(rx_buffer) - 1) {
+        if (i >= sizeof(rx_buffer) - 2) {
```

Comment:
this source file is irrelevant - that's the old MMU2 implementation originating from upstream Marlin

Reply:
I've reverted the patch on mmu2 as it's not being used.